### DISCUSSION TITLE
How to script: "go to buffer-tab number n" (as per tab-bar position)

### DISCUSSION THREAD

**Discussion by metaleap** [orbitalquark/textadept#143] [Q&A] [answered]
I tried this pattern:

```lua
keys['alt+1'] = function() view:goto_buffer(_BUFFERS[0]) end
keys['alt+2'] = function() view:goto_buffer(_BUFFERS[1]) end
keys['alt+3'] = function() view:goto_buffer(_BUFFERS[2]) end
-- etc...
```

as well as:

```lua
keys['alt+1'] = function() view:goto_buffer(0) end
keys['alt+2'] = function() view:goto_buffer(1) end
keys['alt+3'] = function() view:goto_buffer(2) end
-- etc...
```

and finally checking the idx in `for idx, buf in ipairs(_BUFFERS) do ...` and trying both `view:goto_buffer(_BUFFERS[buf])` and `view:goto_buffer(idx)` as per [this find](http://rgieseke.github.io/ta-common/lastbuffer.html), as shown in [L66ff](https://github.com/orbitalquark/textadept/files/7540779/init.lua.txt) ...

but in all 4 attempts, the buffer-tab being chosen jumps all over the place. Could it be that only relative 1/-1 are supported? API doc says: `A buffer or relative buffer number (typically 1 or -1)` so I thought at least the first attempt should do it, as it's passing not a number but a `buffer`.

Alternatively, does a `buffer` have (among its seemingly hundreds) a field yielding its "tab-index/position" or some such? Then I could search through the _BUFFERS instead.

**Comment by orbitalquark** [comment, marked as answer]
`_BUFFERS` is 1-indexed, not 0-indexed. If you start from 1, things should behave as expected.

**Reply by metaleap** [reply]
Oh thanks, it works how I want with `keys['alt+3'] = function() view:goto_buffer(_BUFFERS[3]) end` etc. Could have made a loop but then I'm in an editor with very smooth bulk-editing  :grin:
